discord/commands/options.py

```diff
@@ -197,11 +203,40 @@ def __init__(self, input_type: Any = str, /, description: Optional[str] = None,
         self.min_value: minmax_typehint = kwargs.pop("min_value", None)
         self.max_value: minmax_typehint = kwargs.pop("max_value", None)
 
+        if (input_type != SlashCommandOptionType.integer and input_type != SlashCommandOptionType.number
+                and (self.min_value or self.max_value)):
+            raise AttributeError("Option does not take min_value or max_value if not of type "
+                                 "SlashCommandOptionType.integer or SlashCommandOptionType.number")
+
         if not isinstance(self.min_value, minmax_types) and self.min_value is not None:
             raise TypeError(f'Expected {minmax_typehint} for min_value, got "{type(self.min_value).__name__}"')
         if not (isinstance(self.max_value, minmax_types) or self.min_value is None):
             raise TypeError(f'Expected {minmax_typehint} for max_value, got "{type(self.max_value).__name__}"')
 
+        if self.input_type == SlashCommandOptionType.string:
+            minmax_length_types = (int, type(None))
+        else:
+            minmax_length_types = (type(None))
+        minmax_length_typehint = Optional[Union[minmax_length_types]]
+
+        self.min_length: minmax_length_typehint = kwargs.pop("min_length", None)
+        self.max_length: minmax_length_typehint = kwargs.pop("max_length", None)
+
+        if input_type != SlashCommandOptionType.string and (self.min_length or self.max_length):
```

Comment:
This seems redundant. Merge it with the other if statement above

Reply:
Could you explain? I'm not sure what you mean haha

Reply:
Somewhat above, there's another check that checks if min_length/max_length is set if the input type is not int or number. The problem with this is that technically string isn't int or number, so it would raise even if it's a string.

Reply:
oh